katello-attach-subscription

```diff
@@ -902,8 +906,17 @@ def vdcupdate()
       if @options[:verbose]
         puts "VERBOSE: Start testing facts value for #{sys['name']}"
       end
-      fact_test_passed = KatelloAttachSubscription::HostMatcher.match_host(sys, sub)
+
+      if sub.has_key?('facts')
```

Comment:
`match_host` is also checking for the hostname and the type. these checks are now skipped in the case the sub has no facts defined, is that intentional?

also `match_host` will skip the fact matching anyways if the sub has no facts defined.

Reply:
the checks are done in Line 856 and Line 866

Reply:
for hostname and type, but not for facts

Reply:
sure, but what's the point in making the code more complicated here? executing the checks twice is cheap.

on the other hand, we might as well just call `match_host` once and drop 856/866? (but not in this PR)